### ISSUE TITLE
Overview of variant server

### ISSUE THREAD

**Issue by BoPeng** [vatlab/vat-vs#1]
A variant server is the atomic clock for all component of vat and should be implemented before anything else. Essentially speaking,  a variant server accept a variant and return an unique ID for the variant.

Implementation wise:

1. The variant server should be accessible by any process from a machine:port. It does not have to use TCP/IP.
2. There can be different variant servers that provides variants with different species, reference genome etc.

Protocol:

A: client
B: variant server

A: query info from a port (e.g. `INFO`)
B: return information about the server (e.g. `{'build': `hg19`, `key`: 0x24b5c9}`
A: request a new channels (e.g. `NEW CHANNEL build=hg19`)
B: creates a separate daemon that monitor two channels and return channel IDs (e.g. `9807`, `9808`)
A: send one or more variants to data channel, perhaps in compressed format. (e.g. `chr1\t12341\tA\tC`)
B: receive variants from channel 9807, send ID to 9808
A: read IDs from 9808.

Note that:

1. A variant server should provide a key to identify itself. All components of VAT should use the same variant server to make sure they obtain IDs from the same server.
2. A data provider can in theory work with different variant servers. If we upgrade a variant server and provide a different key, all data providers should be able to re-generate its indexes.

Because of this, we can provide alternative variant server (e.g. accept different reference genome through liftover) and do not have to worry about compatibility with other variant servers.

**Comment by gaow** [comment]
At this stage, are we also planning on supporting indels and CNV (1000 Genomes has both data types to test)?

**Comment by BoPeng** [comment]
Yes. A variant server should accept such variants and return unique IDs. The problems with overlapping variants can be addressed later.
